### PR TITLE
EVG-14416 Skip mergeable check when submitting to the commit queue

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-14416](https://jira.mongodb.org/browse/EVG-14416)

### Description 
There's already a mergeable check in commit queue tasks, so there's no need to check it here. Checking here can make it so that enqueuing very quickly after pushing a commit fails to enqueue

### CODE REVIEW

**Review by Kimchelly — APPROVED** [review]

File: thirdparty/github.go [hunk 1]
```diff
@@ -1026,16 +1026,6 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
...
-	if pr.Mergeable == nil {
-		if *pr.Merged {
-			return pr, errors.New("PR is already merged")
-		}
-		// GitHub hasn't yet tested if the PR is mergeable.
-		// Check back later
-		// See: https://developer.github.com/v3/pulls/#response-1
-		return pr, errors.New("GitHub hasn't yet generated a merge commit")
-	}
-
```
Comment (Kimchelly) [inline thirdparty/github.go:1038]: L1039 needs to check if pr.Mergeable is nil before dereferencing.

**Review by Kimchelly — COMMENTED** [review]
Sorry, I noticed one thing on an unchanged line.

File: thirdparty/github.go (hunk 1 above)
Comment (malikchaya2) [reply thirdparty/github.go:1038]: thanks! switched to utility.FromBoolPtr.

Comment (Kimchelly) [reply thirdparty/github.go:1038]: I think it has to be `utility.FromBoolTPtr` because if the pointer is nil, we're assuming it's mergeable.

Comment (malikchaya2) [reply thirdparty/github.go:1038]: It seems to me that if the pointer is nil, FromBoolPtr returns false, so we end up with !(false), so we enter the if and return errors.New("PR is not mergeable"). 

Am I misreading it?

Comment (Kimchelly) [reply thirdparty/github.go:1038]: I thought the point of removing the nil check was to assume that if the mergeability check isn't available yet, we just skip the check, and would only want to exit early if `*Mergeable` was true. If `nil` (i.e. can't check mergeability yet) is equivalent to `false`, then the if statement will make it claim that it's not mergeable, but we actually don't know if it's mergeable.

Comment (malikchaya2) [reply thirdparty/github.go:1038]: ah yes :see_no_evil: . Thanks! fixed!

File: thirdparty/github.go [hunk 2]
```diff
@@ -1026,7 +1026,7 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
 		return nil, errors.Wrap(err, "GitHub returned an incomplete PR")
 	}
 
-	if !(utility.FromBoolPtr(pr.Mergeable)) {
+	if !(utility.FromBoolTPtr(pr.Mergeable)) {
```
Comment (Kimchelly) [inline thirdparty/github.go:1029]: nit: the parentheses aren't necessary since `!` is a unary operator.

**Review by Kimchelly — APPROVED** [review]
LGTM mod nit.

**Pull request merged**